static/js/popovers.js

```diff
@@ -1124,7 +1124,7 @@ export function register_click_handlers() {
         e.preventDefault();
     });
 
-    $("body").on("click", ".respond_button", (e) => {
+    $("#main_div").on("click", ".respond_button", (e) => {
         // Arguably, we should fetch the message ID to respond to from
```

Comment:
You need to change the code inside handler as well. Currently the message which is selected (surrounded by blue box) is quoted and not where I click. You can see the code for `edit_content_button` to see how we get the correct message.
Also this code should be moved to `click_handlers.js` which contains the edit button click handler code as well.